Submitted By:            Bruce Dubbs <bdubbs_at_linuxfromscratch_dot_org>
Date:                    2012-09-22
Initial Package Version: 10
Upstream Status:         From Upstream
Origin:                  X-Git-Url: http://git.kernel.org/?p=utils%2Fkernel%2Fkmod%2Fkmod.git;a=blobdiff_plain;f=testsuite%2Fpath.c;h=c13ff4b6ec7067c7bedf3c496bbcc7bf904b7692;hp=5cb17bb9c1c0d939a2935d431757bcbfa9fec4e8;hb=d6423417778d;hpb=40758602647a3871e6f432b1c59d2a756208d207
Description:             Fixes a double definition when running the testsuite
                         on x86 hosts.

diff --git a/testsuite/path.c b/testsuite/path.c
index 5cb17bb..c13ff4b 100644
--- a/testsuite/path.c
+++ b/testsuite/path.c
@@ -184,16 +184,20 @@ WRAP_2ARGS(FILE*, NULL, fopen, const char*);
 WRAP_2ARGS(int, -1, mkdir, mode_t);
 WRAP_2ARGS(int, -1, access, int);
 WRAP_2ARGS(int, -1, stat, struct stat*);
-WRAP_2ARGS(int, -1, stat64, struct stat64*);
 WRAP_2ARGS(int, -1, lstat, struct stat*);
+#ifndef _FILE_OFFSET_BITS
+WRAP_2ARGS(int, -1, stat64, struct stat64*);
 WRAP_2ARGS(int, -1, lstat64, struct stat64*);
+WRAP_OPEN(64);
+#endif
 
 WRAP_OPEN();
-WRAP_OPEN(64);
 
 #ifdef HAVE___XSTAT
 WRAP_VERSTAT(__x,);
-WRAP_VERSTAT(__x,64);
 WRAP_VERSTAT(__lx,);
+#ifndef _FILE_OFFSET_BITS
+WRAP_VERSTAT(__x,64);
 WRAP_VERSTAT(__lx,64);
 #endif
+#endif
